apps/activity/views.py

```diff
@@ -1,5 +1,184 @@
+
+        return queryset
+
+    def perform_create(self, serializer):
+        plan = serializer.save()
+        self.request.user.plan = plan
+        self.request.user.save()
+
+    @action(methods=['patch'], detail=False, serializer_class=PlanSerializer, url_path="change")
+    def change_plan(self, request, *args, **kwargs):
+        plan = Plan.objects.get(id=request.user.plan.id)
+        plan.steps = request.data["steps"]
+        plan.calories = request.data["calories"]
+        plan.sleep = request.data["sleep"]
+        plan.water = request.data["water"]
+        plan.save()
+        return Response({"success": True}, status.HTTP_200_OK)
+
+    @action(methods=['post'], detail=False, serializer_class=SleepSerializer, url_path="sleep")
+    def create_sleep(self, request, *args, **kwargs):
+        sleep = request.data["sleep"]
+        plan = self.request.user.plan
+        ActivitySleep.objects.create(sleep=sleep, plan=plan)
+        return Response({"success": True}, status.HTTP_201_CREATED)
+
+    @action(methods=['post'], detail=False, serializer_class=WaterSerializer, url_path="water")
+    def create_water(self, request, *args, **kwargs):
+        sleep = request.data["water"]
+        plan = self.request.user.plan
+        ActivityWater.objects.create(water=sleep, plan=plan)
+        return Response({"success": True}, status.HTTP_201_CREATED)
+
+    @action(methods=['post'], detail=False, serializer_class=CreateFoodSerializer, url_path="food")
+    def create_food(self, request, *args, **kwargs):
+        plan = self.request.user.plan
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(plan=plan)
+        return Response({"success": True}, status.HTTP_201_CREATED)
+
+    @action(methods=['delete'], detail=True, serializer_class=None, url_path="delete-food")
+    def delete_food(self, *args, **kwargs):
+        food_id = kwargs.get("pk")
+        food = ActivityFood.objects.get(id=food_id)
+        food.delete()
+        return Response({"success": True}, status.HTTP_200_OK)
+
+    @action(methods=['get'], detail=False, serializer_class=GetAllCaloriesSerializer, url_path="calories")
+    def foods(self, request, *args, **kwargs):
+        foods = ActivityFood.objects.filter(plan=self.request.user.plan, created_at__date=timezone.now()).order_by("-id")
+        foods_serializer = CreateFoodSerializer(foods, many=True)
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        total_calories = serializer.data['all_calories']
+        response_data = {
+            'all_calories': total_calories,
+            'foods': foods_serializer.data
+        }
+        return Response(response_data, status=status.HTTP_200_OK)
+
+    @action(methods=['post'], detail=False, serializer_class=CreateStepsSerializer, url_path="step")
+    def create_step(self, request, *args, **kwargs):
```

Comment:
Numele la func trebuie sa coincida c url_path, numai ca in loc de - pui _
P.S. Uite te rog la toate endpointurile sa fie asa